doc/en/backwards-compatibility.rst

```diff
@@ -3,6 +3,61 @@
 Backwards Compatibility Policy
 ==============================
 
+.. versionadded: 6.0
+
+pytest is actively evolving and is a project that has been decades in the making,
+we keep learning about new and better structures to express different details about testing.
+
+While we implement those modifications we try to ensure an easy transition and don't want to impose unnecessary churn on our users and community/plugin authors.
+
+As of now, pytest considers multipe types of backward compatibility transitions:
+
+a) trivial: APIs which trivially translate to the new mechanism,
+   and do not cause problematic changes.
+
+   We try to support those indefinitely while encouraging users to switch to newer/better mechanisms through documentation.
+
+b) transitional: the old and new API don't conflict
+   and we can help users transition by using warnings, while supporting both for a prolonged time.
+
+   We will only start the removal of deprecated functionality in major releases (e.g. if we deprecate something in 3.0 we will start to remove it in 4.0), and keep it around for at least two minor releases (e.g. if we deprecate something in 3.9 and 4.0 is the next release, we start to remove it in 5.0, not in 4.0).
+
+   When the deprecation expires (e.g. 4.0 is released), we won't remove the deprecated functionality immediately, but will use the standard warning filters to turn them into **errors** by default. This approach makes it explicit that removal is imminent, and still gives you time to turn the deprecated feature into a warning instead of an error so it can be dealt with in your own time. In the next minor release (e.g. 4.1), the feature will be effectively removed.
+
+
+c) true breakage: should only to be considered when normal transition is unreasonably unsustainable and would offset important development/features by years.
+   In addition, they should be limited to APIs where the number of actual users is very small (for example only impacting some plugins), and can be coordinated with the community in advance.
+
+   Examples for such upcoming changes:
+
+   * removal of ``pytest_runtest_protocol/nextitem``
+   * rearranging of the node tree to include ``FunctionDefinition``
+   * rearranging of ``SetupState``
+
+   True breakages must be announced first in an issue containing:
+
+   * Detailed description of the change
+   * Rationale
+   * Expected impact on users and plugin authors
```

Comment:
We should link an existing issue that describes this type of breakage as an example (and is also an exercise to ensure the constructor API is following this protocol).

Reply:
followup in #6324